multisig/simulations/ethereum/vip-019/index.ts

```diff
@@ -0,0 +1,212 @@
+import { checkInterestRate } from "../../../../src/vip-framework/checks/interestRateModel";
+import { forking, pretendExecutingVip } from "../../../../src/vip-framework/index";
+import {
+  COMPTROLLER,
+  MULTISIG,
+  REWARDS_DISTRIBUTOR,
+  USDC,
+  USDC_REWARD_TRANSFER,
+  vip019,
+  vweETH,
+  weETH,
+} from "../../../proposals/ethereum/vip-019";
+import POOL_REGISTRY_ABI from "./abi/PoolRegistry.json";
+import RESILIENT_ORACLE_ABI from "./abi/ResilientOracle.json";
+import REWARD_DISTRIBUTOR_ABI from "./abi/RewardsDistributor.json";
+import COMPTROLLER_ABI from "./abi/comptroller.json";
+import ERC20_ABI from "./abi/erc20.json";
+import VTOKEN_ABI from "./abi/vToken.json";
+
+const { ethereum } = NETWORK_ADDRESSES;
+const WeETH_ORACLE_NON_EQUIVALENCE = "0x660c6d8c5fddc4f47c749e0f7e03634513f23e0e";
+const BLOCKS_PER_YEAR = BigNumber.from(2628000); // assuming a block is mined every 12 seconds
+const LIQUID_STAKED_COMPTROLLER = "0xF522cd0360EF8c2FF48B648d53EA1717Ec0F3Ac3";
+const WEETH_HOLDER = "0x267ed5f71EE47D3E45Bb1569Aa37889a2d10f91e";
+
+interface RiskParameters {
+  borrowCap: string;
+  supplyCap: string;
+  collateralFactor: string;
+  liquidationThreshold: string;
+  reserveFactor: string;
+  initialSupply: string;
+  vTokenReceiver: string;
+}
+
+const riskParameters: RiskParameters = {
+  borrowCap: "750",
+  supplyCap: "7500",
+  collateralFactor: "0.9",
+  liquidationThreshold: "0.93",
+  reserveFactor: "0.20",
+  initialSupply: "2.76191022",
+  vTokenReceiver: MULTISIG,
+};
+
+interface InterestRateModelSpec {
+  vToken: string;
+  kink: string;
+  base: string;
+  multiplier: string;
+  jump: string;
+}
+
+const interestRateModel: InterestRateModelSpec = {
+  vToken: "vweETH_LiquidStakedETH",
+  kink: "0.45",
+  base: "0",
+  multiplier: "0.09",
+  jump: "0.75",
+};
+
+forking(19640453, () => {
+  let resilientOracle: Contract;
+  let poolRegistry: Contract;
+  let vweETHContract: Contract;
+  let comptroller: Contract;
+  let rewardDistributor: Contract;
+  let usdc: Contract;
+  let nonEquivalenceOracle: Contract;
+
+  before(async () => {
+    resilientOracle = await ethers.getContractAt(RESILIENT_ORACLE_ABI, ethereum.RESILIENT_ORACLE);
+    poolRegistry = await ethers.getContractAt(POOL_REGISTRY_ABI, ethereum.POOL_REGISTRY);
+    vweETHContract = await ethers.getContractAt(VTOKEN_ABI, vweETH);
+    comptroller = await ethers.getContractAt(COMPTROLLER_ABI, COMPTROLLER);
+    rewardDistributor = await ethers.getContractAt(REWARD_DISTRIBUTOR_ABI, REWARDS_DISTRIBUTOR);
+    usdc = await ethers.getContractAt(ERC20_ABI, USDC);
+    nonEquivalenceOracle = await ethers.getContractAt(RESILIENT_ORACLE_ABI, WeETH_ORACLE_NON_EQUIVALENCE);
+  });
+
+  describe("Pre-VIP behavior", () => {
+    it("check price", async () => {
+      await expect(resilientOracle.getPrice(weETH)).to.be.reverted;
+    });
+  });
+
+  describe("Post-VIP behavior", async () => {
```

Comment:
We can add a generic check for VToken using `src/vip-framework/checks/checkVToken.ts`